File: Python/Endpoint Examples/JSON Payload/encrypted-pdf.py
import requests
import json

with open('/path/to/file', 'rb') as f:
    upload_data = f.read()

print("Uploading file...")
upload_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_data,
                    headers={'Content-Type': 'application/octet-stream', 'Content-Filename': 'file.pdf', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Upload response status code: " + str(upload_response.status_code))

if upload_response.ok:
    upload_response_json = upload_response.json()
    print(json.dumps(upload_response_json, indent = 2))


    uploaded_id = upload_response_json['files'][0]['id']
    encrypt_data = { "id" : uploaded_id, 'new_open_password': 'password' }
    print(json.dumps(encrypt_data, indent = 2))

    print("Processing file...")
    encrypt_response = requests.post(url='https://api.pdfrest.com/encrypted-pdf',
                        data=json.dumps(encrypt_data),
                        headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



    print("Processing response status code: " + str(encrypt_response.status_code))
    if encrypt_response.ok:
        encrypt_response_json = encrypt_response.json()
        print(json.dumps(encrypt_response_json, indent = 2))

    else:
        print(encrypt_response.text)
else:
    print(upload_response.text)

File: DotNET/Endpoint Examples/JSON Payload/encrypted-pdf.cs

using Newtonsoft.Json.Linq;
using System.Text;

// Toggle deletion of sensitive files (default: false)
var deleteSensitiveFiles = false;

using (var httpClient = new HttpClient { BaseAddress = new Uri("https://api.pdfrest.com") })
{
    using (var uploadRequest = new HttpRequestMessage(HttpMethod.Post, "upload"))
    {
        uploadRequest.Headers.TryAddWithoutValidation("Api-Key", "xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx");
        uploadRequest.Headers.Accept.Add(new("application/json"));

        var uploadByteArray = File.ReadAllBytes("/path/to/file");
        var uploadByteAryContent = new ByteArrayContent(uploadByteArray);
        uploadByteAryContent.Headers.TryAddWithoutValidation("Content-Type", "application/octet-stream");
        uploadByteAryContent.Headers.TryAddWithoutValidation("Content-Filename", "filename.pdf");


        uploadRequest.Content = uploadByteAryContent;
        var uploadResponse = await httpClient.SendAsync(uploadRequest);

        var uploadResult = await uploadResponse.Content.ReadAsStringAsync();

        Console.WriteLine("Upload response received.");
        Console.WriteLine(uploadResult);

        JObject uploadResultJson = JObject.Parse(uploadResult);
        var uploadedID = uploadResultJson["files"][0]["id"];
        using (var encryptRequest = new HttpRequestMessage(HttpMethod.Post, "encrypted-pdf"))
        {
            encryptRequest.Headers.TryAddWithoutValidation("Api-Key", "xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx");
            encryptRequest.Headers.Accept.Add(new("application/json"));

            encryptRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");


            JObject parameterJson = new JObject
            {
                ["id"] = uploadedID,
                ["new_open_password"] = "password"
            };

            encryptRequest.Content = new StringContent(parameterJson.ToString(), Encoding.UTF8, "application/json"); ;
            var encryptResponse = await httpClient.SendAsync(encryptRequest);

            var encryptResult = await encryptResponse.Content.ReadAsStringAsync();

            Console.WriteLine("Processing response received.");
            Console.WriteLine(encryptResult);
            JObject encryptResultJson = JObject.Parse(encryptResult);
            var outputID = encryptResultJson["outputId"];

            // All files uploaded or generated are automatically deleted based on the 
            // File Retention Period as shown on https://pdfrest.com/pricing. 
            // For immediate deletion of files, particularly when sensitive data 
            // is involved, an explicit delete call can be made to the API.
            //
            // Deletes all files in the workflow, including outputs. Save all desired files before enabling this step.

            if (deleteSensitiveFiles)
            {
                using (var deleteRequest = new HttpRequestMessage(HttpMethod.Post, "delete"))
                {
                    deleteRequest.Headers.TryAddWithoutValidation("Api-Key", "xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx");
                    deleteRequest.Headers.Accept.Add(new("application/json"));
                    deleteRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");

                    JObject deleteJson = new JObject
                    {
                        ["ids"] = $"{uploadedID}, {outputID}"
                    };
                    deleteRequest.Content = new StringContent(deleteJson.ToString(), Encoding.UTF8, "application/json");
                    var deleteResponse = await httpClient.SendAsync(deleteRequest);
                    var deleteResult = await deleteResponse.Content.ReadAsStringAsync();
                    Console.WriteLine(deleteResult);
                }
            }
        }
    }
}
